File: 121.best-time-to-buy-and-sell-stock.py
# @leet start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 0
        sol = 0
        while l < len(prices) and r < len(prices):
            if prices[l] > prices[r]:
                l = r
            temp= prices[r] - prices[l] 
            if temp> sol:
                sol = temp
            r += 1
        return sol


# @leet end

File: 121.best-time-to-buy-and-sell-stock.java
// @leet start
class Solution {
    public int maxProfit(int[] prices) {
		int l = 0;
		int r = 0;
		int sol =0;
		while (l < prices.length && r< prices.length) {
			if (prices[l]>prices[r]) l=r;
			if(prices[r]-prices[l] > sol) sol= prices[r]-prices[l];
			r++;
		}
		return sol;
    }
}
// @leet end
